pkg/bootkube/create.go

```diff
@@ -144,6 +163,50 @@ func createAssets(manifestDir string) error {
 	return nil
 }
 
+func waitForCRDRollout(client resource.RESTClient, gvk schema.GroupVersionKind, obj runtime.Object) error {
+	return wait.PollImmediate(crdRolloutDuration, crdRolloutTimeout, func() (bool, error) {
+		var crd v1beta1ext.CustomResourceDefinition
+		bytes, err := json.Marshal(obj)
+		if err != nil {
+			return false, err
+		}
+		if err = json.Unmarshal(bytes, &crd); err != nil {
+			return false, err
+		}
+		uri := customResourceDefinitionKindURI(crd.Spec.Group, crd.Spec.Version, crd.GetNamespace(), crd.Spec.Names.Singular)
+		res := client.Get().RequestURI(uri).Do()
+		if res.Error() != nil {
+			if apierrors.IsNotFound(res.Error()) {
+				return false, nil
+			}
+			return false, res.Error()
+		}
+		return true, nil
+	})
+}
+
+// customResourceDefinitionKindURI returns the URI for the CRD kind.
+//
+// Example of apiGroup: "tco.coreos.com"
+// Example of version: "v1"
+// Example of namespace: "default"
+// Example of resourceKind: "ChannelOperatorConfig"
+func customResourceDefinitionKindURI(apiGroup, version, namespace, resourceKind string) string {
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
+	plural, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{
```

Comment:
This is kind of scary description: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/restmapper.go#L145

Reply:
Yeah, definitely agreed, however we've been using that function within operator-client (where this is taken from) without issue for a while now. Before that we were using `KindToResource` before it was removed in favor of this scarier named function.